packages/sdk-react/src/components/counterparts/CounterpartsTable/CounterpartsTable.tsx

```diff
@@ -355,6 +366,46 @@ const CounterpartsTableBase = ({
     return <AccessRestriction />;
   }
 
+  if (!isLoading && counterparts?.data.length === 0) {
+    return (
+      <DataGridEmptyState
+        icon={
+          <NoDataIcon sx={{ fontSize: '4rem', color: 'primary.main', mb: 2 }} />
+        }
+        title={t(i18n)`No Counterparts`}
+        descriptionLine1={t(i18n)`You don’t have any counterparts yet.`}
+        descriptionLine2={t(i18n)`You can create your first counterpart.`}
+        actionOptions={[t(i18n)`Organization`, t(i18n)`Individual`]}
+        actionButtonLabel={t(i18n)`Create new`}
+        onAction={(action) => {
+          if (!setType) return;
+          if (action === t(i18n)`Organization`) {
+            setType('organization');
+          } else if (action === t(i18n)`Individual`) {
+            setType('individual');
+          }
+        }}
+        type="no-data"
+      />
+    );
+  }
+
+  if (isError) {
+    return (
+      <DataGridEmptyState
+        icon={
+          <ErrorIcon sx={{ fontSize: '4rem', color: 'error.main', mb: 2 }} />
+        }
+        title={t(i18n)`Failed to Load Counterparts`}
+        descriptionLine1={t(i18n)`There was an error loading counterparts.`}
+        descriptionLine2={t(i18n)`Please try again later.`}
+        actionButtonLabel={t(i18n)`Reload`}
+        onAction={() => window.location.reload()}
```

Comment:
While this approach is fine for <iframe> integration, it may have unexpected consequencies when our components were integrated as react components. I suggest to use useQuery().refetch instead:
`onAction={() => refetch()}`